### ISSUE TITLE
ideas for next version of `ndtiff` reader

### ISSUE THREAD

**Issue by ieivanov** [czbiohub-sf/iohub#183]
Writing down some ideas for the next version of `ndtiff` reader (as part of the UAPI?) before I forget them.

In the reader should convert the `Dataset.axes` from `SortedSet` to `tuple`. The difference is largely that the values in `SortedSet` are unique, while values in a tuple may not be. We can then index into these axes.

We should update `_check_coordinates` to take integers or strings and return:
- `None` if the axis index is 0 and the axis is not part of the dataset
- string index if the axis is string-valued (currently this only works if the axis index is 0 because we cannot index into the SortedSet; converting the axis indices to tuple will solve this issue)
- raise `ValueError` if an image at these coordinates does not exist (using `Dataset.has_image`); we can then catch this error during conversion, for example; this could also be a separate function or implemented in `get_image` and the like.

This will allow us to loop through integer indices during conversion and proceed safely if parts of the dataset do not exist. Note that `Dataset.axes` contains the largest set of possible axis indices. For example, given

```python
axes = {
    "position": ['Pos0', 'Pos1'],
    "time": [0, 1, 2]
}
```

`has_image(position='Pos1', time=1)` may be `False` if `has_image(position='Pos0', time=1)` is `True`

**Comment by ziw-liu** [comment]
@ieivanov Is this still relevant after #222?

**Comment by ieivanov** [comment]
I think this has been resolved, closing.
